### PR TITLE
switched to scalismo 0.16.0-RC1

### PR DESCRIPTION
**Pull request by BernhardEgger**
The newest scalismo release fixes issues with random number generation in the background:
https://github.com/unibas-gravis/scalismo/issues/222
Since random numbers are somehow essential for a lot of applications we have we should update...
Besides this some interfaces changed.
I'm aware that it is only a release candidate of scalismo, we should wait for a stable version for our next release of scalismo-faces.
I'm actually unsure if we should use UnstructuredPointsDomain (as it is done in scalimso) or DiscreteDomain (as it was in the MoMoIO Line 309)???
However to test I rendered an instance of the Model and it works.

### CODE REVIEW

File: src/main/scala/scalismo/faces/momo/PancakeDLRGP.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ import scalismo.utils.Random
   * @tparam D dimensionality of domain
   * @tparam Value value type of model
   */
-case class PancakeDLRGP[D <: Dim: NDSpace, Value](gpModel: DiscreteLowRankGaussianProcess[D, Value],
+case class PancakeDLRGP[D <: Dim: NDSpace, DDomain <: DiscreteDomain[D], Value](gpModel: DiscreteLowRankGaussianProcess[D, DDomain, Value],
```
Comment (Andreas-Forster) [inline src/main/scala/scalismo/faces/momo/PancakeDLRGP.scala]: We should allow here also covariant classes of `DDomain`. So we should have the types as `[D <: Dim: NDSpace, +DDomain <: DiscreteDomain[D], Value]` . When changing we need also to replace some of the usage of `DDomain` with `DiscreteDomain[D]` in the rest of the code. I think the nescessary changes are in the function definitions of `pdf`, `logpdf`, `project` and `coefficients`.

**Review by Andreas-Forster — CHANGES_REQUESTED** [review]
I am not yet sure if we should go with the `UnstructeredPointsDomain`. Here I need more time to figure it our. My first thought would be to use the `DiscreteDomain` interface.

For the changes in the PancakeDLRGP I have left a comment where I think you should change the type to be covariant.

**Comment by Andreas-Forster** [comment]
I discussed the point of which domain type to use with Marcel from scalismo. We agreed in the end that it is best to use a domain as specific as possible. So I suggest we go with the UnstructuredPointsDomain.

So only the small change for the covariant type parameter is needed for my approve for this PR.

Why can I edit your comments?

**Comment by BernhardEgger** [comment]
Thanks for discussing this with Marcel!!!
I will update the remaining Issues...
Why can I edit your comments?

**Review by Andreas-Forster — APPROVED** [review]

**Pull request merged**